File: NEETCODE/Easy/is-anagram.js
// https://neetcode.io/problems/is-anagram

class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        const lenS = s.length;
        if (lenS !== t.length) {
            return false
        }

        const arr = new Array(26).fill(0);
        for (let i=0; i < lenS; i++) {
            arr[s.charCodeAt(i) - 'a'.charCodeAt(0)]++;
            arr[t.charCodeAt(i) - 'a'.charCodeAt(0)]--;
        }

        return arr.every(num => num === 0)
    }
}

File: NEETCODE/Easy/is-anagram.py
# https://neetcode.io/problems/is-anagram

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_list = sorted(s)
        t_list = sorted(t)

        if len(s) != len(t):
            return False
        
        return s_list == t_list